Bonsai.Design/ObjectTextVisualizer.cs

```diff
@@ -30,27 +31,90 @@ public class ObjectTextVisualizer : BufferedVisualizer
         /// <inheritdoc/>
         protected override void ShowBuffer(IList<Timestamped<object>> values)
         {
-            if (values.Count > 0)
+            if (values.Count == 0)
+                return;
+
+            var sb = new StringBuilder();
+
+            // Trim old values if bufferSize was reduced
+            while (buffer.Count >= bufferSize)
+                buffer.Dequeue();
+
+            // Add new values to the buffer (and only the ones which might appear)
+            for (int i = Math.Max(0, values.Count - bufferSize); i < values.Count; i++)
             {
-                base.ShowBuffer(values);
-                textBox.Text = string.Join(Environment.NewLine, buffer);
-                textPanel.Invalidate();
+                var rawText = values[i].Value?.ToString() ?? string.Empty;
+                sb.Clear();
+                sb.EnsureCapacity(rawText.Length);
+                MakeDisplayString(rawText, sb);
+                DisplayString(sb.ToString());
```

Comment:
```suggestion
                sb.Clear();
                AppendDisplayText(sb, values[i].Value);
                AppendString(sb.ToString());
```

Given all the above, I guess this bit would look like this. I was wondering here whether we will ever want to provide the timestamp, but I guess if we ever need that we can later make a non-breaking change by introducing a new virtual method `AppendTimestampedDisplayText` which receives the timestamped value and by default just calls `AppendDisplayText` with the value.

Reply:
Yeah, providing the timestamp to `AppendDisplayText` just opens another weird thing to handle in the `Show(object)` edge case.

I think if we're making a text visualizer that also shows timestamps then at that point we need to revisit making this visualizer have configuration.